test.env

```diff
@@ -21,3 +21,4 @@ export OCI_KEY_FILE=tools/c7n_oci/tests/data/oci_api_key.pem
 export OCI_REGION=us-ashburn-1
 export OCI_TENANCY=ocid1.tenancy.oc1..unique_ID
 export OCI_LOG_COMPARTMENT_ID="ocid1.test.logging.compartment.id"
+export OCI_INSTANCE_PRINCIPAL=no
```

Comment:
where is this environment variable consumed, Ic OCI_CLI_AUTH later in the code?

Reply:
This variable is to control when to run the instance principal functional test. Not used for anything outside of that.

Reply:
environment variables to control a single test suite's behavior are kinda of broken.  as is oci already has more than 2x the number of test env vars of any other provider. at min due to the volume and that these are many non standard they should be documented in the test.env file.

Reply:
For the purposes of testing I am okay to auto detect instance auth and then only run the test, thus avoiding the env. var. Let me know if that is acceptable.